Rename 'activity' to 'element' in all records

ThorbenLindhauer opened the pull request:
- workflow instance record
- incident record
- job headers
- message subscription
- workflow subscription

closes #1528

menski commented:
The elasticsearch exporter mappings have to be adjusted to the changes, see https://github.com/zeebe-io/zeebe/tree/1528-element-id/exporters/elasticsearch-exporter/src/main/resources

menski commented:
Reminder we have to regenerate the Go gRPC stub and mocks as this is not done automatically. See for instructions:
- https://github.com/zeebe-io/zeebe/blob/develop/gateway-protocol/README.md
- https://github.com/zeebe-io/zeebe/blob/develop/clients/go/README.md

In case you need help or want me to do it just ping me

saig0 commented on broker-core/src/main/java/io/zeebe/broker/workflow/processor/job/JobCompletedEventProcessor.java:
```diff
@@ -44,7 +44,7 @@ public void processRecord(
     final JobRecord jobEvent = record.getValue();
     final JobHeaders jobHeaders = jobEvent.getHeaders();
-    final long activityInstanceKey = jobHeaders.getActivityInstanceKey();
+    final long activityInstanceKey = jobHeaders.getElementInstanceKey();
```
```suggestion
    final long elementInstanceKey = jobHeaders.getElementInstanceKey();
```

saig0 commented on broker-core/src/main/java/io/zeebe/broker/workflow/processor/job/JobCreatedProcessor.java:
```diff
@@ -41,7 +41,7 @@ public void processRecord(
 
     final JobHeaders jobHeaders = record.getValue().getHeaders();
-    final long activityInstanceKey = jobHeaders.getActivityInstanceKey();
+    final long activityInstanceKey = jobHeaders.getElementInstanceKey();
```
```suggestion
    final long elementInstanceKey = jobHeaders.getElementInstanceKey();
```

saig0 commented on broker-core/src/main/java/io/zeebe/broker/workflow/processor/servicetask/TerminateServiceTaskHandler.java:
```diff
@@ -50,8 +50,8 @@ protected void addTerminatingRecords(
           .setWorkflowInstanceKey(activityInstanceEvent.getWorkflowInstanceKey())
-          .setActivityId(activityInstanceEvent.getActivityId())
-          .setActivityInstanceKey(activityInstance.getKey());
+          .setElementId(activityInstanceEvent.getElementId())
```
```suggestion
          .setElementId(elementInstanceEvent.getElementId())
```

saig0 commented on clients/java/src/main/java/io/zeebe/client/api/response/JobHeaders.java:
```diff
@@ -31,8 +31,8 @@
   long getWorkflowKey();
 
   /** @return id of the activity */
```
```suggestion
  /** @return id of the element */
```

saig0 commented on clients/java/src/main/java/io/zeebe/client/api/response/JobHeaders.java:
```diff
@@ -31,8 +31,8 @@
-  String getActivityId();
+  String getElementId();
 
   /** @return key of the activity instance */
```
```suggestion
  /** @return key of the element instance */
```

saig0 commented on gateway/src/main/java/io/zeebe/gateway/api/record/IncidentRecord.java:
```diff
@@ -38,13 +38,13 @@
    * @return the id of the activity this incident belongs to. Can be <code>null</code> if the
```
```suggestion
   * @return the id of the element this incident belongs to. Can be <code>null</code> if the
```

saig0 commented on gateway/src/main/java/io/zeebe/gateway/api/record/IncidentRecord.java:
```diff
@@ -38,13 +38,13 @@
+  String getElementId();
 
   /**
    * @return the key of the activity instance this incident belongs to. Can be <code>null</code> if
```
```suggestion
   * @return the key of the element instance this incident belongs to. Can be <code>null</code> if
```

saig0 commented on gateway/src/main/java/io/zeebe/gateway/api/record/WorkflowInstanceRecord.java:
```diff
@@ -39,7 +39,7 @@
   /**
    * @return the id of the current activity, or empty if the event does not belong to an activity.
```
```suggestion
   * @return the id of the current element, or empty if the event does not belong to an activity.
```

saig0 commented on gateway/src/main/java/io/zeebe/gateway/impl/command/IncidentCommandImpl.java:
```diff
@@ -50,9 +50,9 @@ public String toString() {
     builder.append(", workflowInstanceKey=");
     builder.append(getWorkflowInstanceKey());
     builder.append(", activityId=");
```
```suggestion
    builder.append(", elementId=");
```

saig0 commented on gateway/src/main/java/io/zeebe/gateway/impl/command/WorkflowInstanceCommandImpl.java:
```diff
@@ -63,7 +63,7 @@ public String toString() {
     builder.append(", version=");
     builder.append(getVersion());
     builder.append(", activityId=");
```
```suggestion
    builder.append(", elementId=");
```

saig0 commented on gateway/src/main/java/io/zeebe/gateway/impl/event/IncidentEventImpl.java:
```diff
@@ -50,9 +50,9 @@ public String toString() {
     builder.append(", workflowInstanceKey=");
     builder.append(getWorkflowInstanceKey());
     builder.append(", activityId=");
```
```suggestion
    builder.append(", elementId=");
```

saig0 commented on gateway/src/main/java/io/zeebe/gateway/impl/event/WorkflowInstanceEventImpl.java:
```diff
@@ -51,7 +51,7 @@ public String toString() {
     builder.append(", version=");
     builder.append(getVersion());
     builder.append(", activityId=");
```
```suggestion
    builder.append(", elementId=");
```

GitCop commented:
There were the following issues with your Pull Request

 * Commit: 5833cebad35ae1691e29dffc5cd7991f4cdac7fa
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

Guidelines are available at https://github.com/zeebe-io/zeebe/blob/develop/CONTRIBUTING.md

---
This message was auto-generated by https://gitcop.com

ThorbenLindhauer commented:
@saig0

I have made the following changes:

- updated the locations that you marked
- re-generated the go client and adapted where necessary
- updated the debug http exporter (was @menski actually :)

ThorbenLindhauer commented:
bors r+
